lib/record_store/provider/ns1.rb

```diff
@@ -0,0 +1,192 @@
+require 'ns1'
+
+module RecordStore
+  class Provider::NS1 < Provider
+    class << self
+      def client
+        Provider::NS1::Client.new(api_key: secrets['api_key'])
+      end
+
+      # Downloads all the records from the provider.
+      #
+      # Returns: an array of `Record` for each record in the provider's zone
+      def retrieve_current_records(zone:, stdout: $stdout)
+        records_for_zone(zone)
+         .map{|short_record| build_from_api(short_record, zone)}
+         .flatten
+      end
+
+      # Returns an array of the zones managed by provider as strings
+      def zones
+        client.zones.map do |zone|
+          zone["zone"]
+        end
+      end
+
+      private
+
+      # Fetches simplified records for the provided zone
+      def records_for_zone(zone)
+        client.zone(zone)["records"]
+      end
+
+      # Creates a new record to the zone. It is expected this call modifies external state.
+      #
+      # Arguments:
+      # record - a kind of `Record`
+      def add(record, zone)
+        new_answers = [{ answer: [ build_api_answer_from_record(record) ]}]
```

Comment:
Style: spacing

```ruby
        new_answers = [{ answer: [ build_api_answer_from_record(record) ] }]
```

Reply:
In lieu of a linter (which would be something else to handle in a follow-on PR), please provide a suggestion, @XaF .  This is a rare case where Github's border-line-useless single line suggestions would work out 😂

Reply:
Updated my code as a suggestion (had to google to find out how.... and this is as easy as saying the code is of `suggestion` type...)
Followed what rubocop does for `spy`: no spaces in list, spaces in hash

Reply:
It has been updated as you suggested, thanks :)